src/core/js/tracking.js

```diff
@@ -71,14 +71,17 @@ define([
 
             // Assessment completed required.
             if (this._config._requireAssessmentCompleted) {
-                if (!this._assessmentState) {
+                if (!this._assessmentState && !Adapt.offlineStorage.get('assessment')) {
```

Comment:
This code should not use internal assessment storage mechanisms.

Reply:
Should we not have an "assessment:restored" event instead?

Reply:
We probably should, but then are we talking about having the core listening out for a plugin event?

Reply:
Yea, i think it's a bit cleaner to have `assessment:complete` and `assessment:restored` rather than having to check for assessment specific storage stuff though?